Add more docblocks and fix XAUTOCLAIM response handler.

michael-grunder opened the pull request:
* Documentation:  Many more docblocks.
* Modify the `XCLAIM` reply handler so it can also consume an `XAUTOCLAIM` reply, which has a different structure.

michael-grunder commented:
The `XAUTOCLAIM` test handler actually segfaults in `RedisCluster` but I will figure that out tomorrow.\

Edit:  Segfault fixed, but the cluster handler will need to be reworked.

yatsukhnenko commented on library.c:
```diff
@@ -2023,6 +2021,7 @@ redis_read_xclaim_response(RedisSock *redis_sock, int count, zval *rv) {
 
             add_next_index_stringl(rv, id, li);
             efree(id);
+            id = NULL;
```
I think er need to move this to a begin of `for` loop because there is `free` also in `else` block

michael-grunder replied:
Good catch, moved.

michael-grunder committed:
Add more docblocks and fix XAUTOCLAIM response handler.

- Finish adding docblocks with examples for all of the stream commands.
- Fix XAUTOCLAIM response handler (the reply has a slightly different
  structure to XCLAIM.
